### ISSUE TITLE
Default to single literal type whenever value is omitted

### ISSUE THREAD

**Issue by prmph** [Microsoft/TypeScript#20500]
Consider the interface:
```
interface xyz {
  num: 1
  str: "ok"
}

```
Currently the following snippet generates a type error:
`let foo: xyz = {}`

My proposal is the whenever a variable is declared to contain only one possible value, that value should be assumed whenever the variable's value is not specified. So, the above snippet would then not generate an error, but this would:
`let foo: xyz = { num: 2 }`

This proposed feature will make it easier to write clearer and simpler TypeScript code.

**Comment by andy-ms** [comment]
That would require changing the semantics of JavaScript. See the [design goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by prmph** [comment]
I don't quite understand what is meant "that would change the semantics of JavaScript". All TypeScript is a type checker; what I mean is that for purposes of type checking, the variable should be assumed to be of the proper type if it does not have a value assigned. This should not have any effect on the code generated.

In fact what I'm proposal is already done in TypeScript to some extent. Consider this code snippet
```
let xyz = class {
  x: string
  y: "ok"
  z: number
}
let ok = new xyz().y
```
This code does not generate an error, although `new xyz().y` , and hence `ok` do not have any value assigned

**Comment by andy-ms** [comment]
Oh, I thought you meant to fill the value in.
I don't wee why we should allow you to omit a field of a literal type, though. If you access the field you will get an error at runtime because you get `undefined` instead of the value you were expecting.
As for the second error you show, see #20166.
